src/LiveComponent/src/LiveComponentHydrator.php

```diff
@@ -66,6 +66,7 @@ public function dehydrate(object $component, ComponentAttributes $attributes, Li
 
         $dehydratedProps = new DehydratedProps();
         foreach ($componentMetadata->getAllLivePropsMetadata() as $propMetadata) {
+            $propMetadata = $propMetadata->withModifier($component);
```

Comment:
This is the riskiest part of this PR. Currently, if you want the `LivePropMetadata[]` for a component, there is one source of truth: `getAllLivePropsMetadata()`. But now, someone needs to know to loop over these `LivePropMetadata` and call `withModifier($component)` to get the final version. That bothers me :/. I understand the "chicken and egg" problem, though: we need to hydrate the component (or as much of it as possible) before we can execute the modifier callbacks. I wish there was a way around that, but I can't think of anything more clever than your sorting algo.

However, to minimize the problem I mentioned above, would it be possible to add an `object $component` argument to `getAllLivePropsMetadata()` and make it return an iterable? Internally, it would do the sorting, then, one-by-one, call `->withModifier()` and yield the final `LiveProp`. Would that work? Then everywhere in the code would call `$componentMetadata->getAllLivePropsMetadata($component)`.